strapi/strapi
# Fix/i18n locale queries forbidden access

### What does it do?

Require custom scopes for Query.i18nLocale and Query.i18nLocales so that it stays consistent with what we currently have for the REST content API.

Note: Currently, read-only tokens will fail to have access to the routes requiring the listLocale scope because it doesn't end with find or findOne. There are ongoing discussions on how to fix this issue.

### Why is it needed?

Stay consistent between the REST and GraphQL APIs

### How to test it?

- Create a UP user
- Head to /graphql
- Make a query to fetch the list of locales > It should fail with a forbidden error
- Set the permission to listLocales to your user
- Make the same query to get the list of locales > It should work and returns the data

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/13653
